clients/rospy/src/rospy/impl/tcpros_base.py

```diff
@@ -737,8 +750,8 @@ def _reconnect(self):
             except TransportInitError:
                 self.socket = None
                 
-            if self.socket is None:
-                # exponential backoff
+            if self.socket is None and interval < 30.:
```

Comment:
I don't mind either or. The ROS 1 code base has pretty bad style and I don't see the way a float is written as important enough to spend more time on it.